frontend/src/components/EditProfile.js

```diff
@@ -0,0 +1,236 @@
+import React, { useState, useEffect } from 'react'
+import { useSelector, useDispatch, batch } from 'react-redux'
+import {
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  Button,
+  FormLabel,
+  Input,
+  Stack,
+  InputGroup,
+  InputLeftElement,
+  Icon,
+  useToast,
+  Text,
+} from '@chakra-ui/react'
+import { FaUserEdit } from 'react-icons/fa'
+import { MdOutlineEdit } from 'react-icons/md'
+import { AtSignIcon } from '@chakra-ui/icons'
+import { RiLockPasswordFill } from 'react-icons/ri'
+
+import { API_URL_USER } from '../utils/urls'
+import DeleteUserAccountBtn from './DeleteUserAccountBtn'
+import user from '../reducers/user'
+
+const EditProfile = ({ isOpen, onClose }) => {
+  const userProfile = useSelector((store) => store.user)
+  const userId = useSelector((store) => store.user.userId)
+  const errorMessage = useSelector((store) => store.user.error)
+
+  const [firstname, setFirstname] = useState()
+  const [lastname, setLastname] = useState()
+  const [email, setEmail] = useState()
+  const [password, setPassword] = useState()
+  const [accessToken, setAccessToken] = useState()
+
+  const dispatch = useDispatch()
+  const toast = useToast()
+
+  // When we have data in the userProfile -> set firstname, lastname, email and accessToken
+  useEffect(() => {
+    setFirstname(userProfile.firstname)
+    setLastname(userProfile.lastname)
+    setEmail(userProfile.email)
+    setAccessToken(userProfile.accessToken)
```

Comment:
That's one way to do it - in other approach, you could initialise state properties with values from redux store, like this :

```
const [firstname, setFirstname] = useState(userProfile.firstname)
```